src/Migrate/custom/Set-AzMigrateServerReplication.ps1

```diff
@@ -527,10 +527,49 @@ function Set-AzMigrateServerReplication {
             }
 
             if ($HasDiskToUpdate) {
+                $diskIdDiskTypeMap = @{}
+                $previousOsDiskId
```

Comment:
You don't need to declare variables before using them in PowerShell. Also the scope level doesn't matter here.